domino-ui/src/main/java/org/dominokit/domino/ui/tree/TreeItem.java

```diff
@@ -798,6 +798,14 @@ public void setValue(T value) {
     this.value = value;
   }
 
+  /** Clear all direct children of the item, also remove the item element from the DOM tree */
+  public void clear() {
+    subItems.stream().forEach(TreeItem::clear);
+    subItems.clear();
+
+    super.remove();
+  }
+
```

Comment:
What I would expect from a clear method is to remove the child elements/items but not to remove the element itself.
While the remove method is expected to remove the item and all of its sub-items.

Reply:
I did this because that the remove method in TreeItem is already overriding something.

Change clear method in TreeItem to clearAndRemove? Or you have other approaching to clear a Tree?

Reply:
I would expect the clear method to actually call remove on all sub-items. I think by just doing that it will function correctly.

Reply:
![image](https://user-images.githubusercontent.com/23259391/172136876-ac09af89-2a9a-4357-885f-77efe7904d49.png)

If I call remove on sub-items, which are also a TreeItem instance. You can see that they are going to call the parent TreeItem again in their remove method. I am afraid for an unnecessary bad performance and possible concurrentmodificationexception.

The parent TreeItem is doing a loop in clear method, while the child TreeItem is calling parent TreeItem again to remove it?

Reply:
Other option is just do not override remove method in TreeItem, since we already have removeItem method there.

At this moment, you already see that removeItem is not efficient implemented.
`subItems.remove(item);
item.remove();`

The item.remove() is actually doing only super.remove() stuff.

Reply:
I am aware of the issue and this might need to be revisited, but the key here is recursion and maybe immutability. I think this is why we had a wrapping list before for the sub-items. :thinking: